### PR TITLE
HomeTask2

### PR DESCRIPTION
**Pull request by LeraShagoyko**

### CODE REVIEW

File: Lecture2/task1/Vegetable.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+abstract public class Vegetable {
```
Comment (acherkas) [inline Lecture2/task1/Vegetable.java]: first we should set access modifier (public,private..etc)
public abstract class ...

File: Lecture2/task1/Vegetable.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    protected int calories;
+
```
Comment (acherkas) [inline Lecture2/task1/Vegetable.java:4]: add weight field for more readable logic

File: Lecture2/task1/Vegetable.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    public int getCalories() {
+        return calories;
+    }
+
```
Comment (acherkas) [inline Lecture2/task1/Vegetable.java:13]: add constructor
public Vegetable(int weight, int calories) {
        this.weight = weight;
        this.calories = calories;
    }

File: Lecture2/task1/Salad.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+public class Salad {
+    private String name;
+    private TreeMap<Integer, Vegetable> vegetables = new TreeMap<>();
```
Comment (acherkas) [inline Lecture2/task1/Salad.java]: please read about the Maps and be sure that you always can have the unique keys
so the following construction won't be valid:
vegetables.put(1, Vegetable);
vegetables.put(1, Vegetable);

please choose another logic for adding Vegetables
for example List<Vegetable>

File: Lecture2/task1/Vegetable.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+abstract public class Vegetable {
+    protected int calories;
```
Comment (acherkas) [inline Lecture2/task1/Vegetable.java]: why is protected?

File: Lecture2/task2/Child.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+public class Child {
+    protected int id;
```
Comment (acherkas) [inline Lecture2/task2/Child.java]: why protected?

File: Lecture2/task2/Application2.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public static Map<Integer, String> convertListToMap(List<Child> listForConverting) {
+        Map<Integer, String> newMap = listForConverting.stream()
+                .collect(Collectors.toMap(Child::getId, Child::getLastName));
+        System.out.println(newMap);
+        return newMap;
+    }
+
+    public static List<Child> createNewList() {
```
Comment (acherkas) [inline Lecture2/task2/Application2.java]: please add the size of generated items in parameter

File: Lecture2/task2/Application2.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        Map<Integer, String> newMap = convertListToMap(originList);
+
+        Map<Integer, String> mapFilteredById = filteredById(newMap);
+        System.out.println(mapFilteredById);
+
+        Collections.sort(filteredChildren, new ChildAgeComparator());
+        System.out.println(filteredChildren);
+    }
+
+    public static List<Child> filterChildrenByLastName(List<Child> notFilteredChildren) {
```
Comment (acherkas) [inline Lecture2/task2/Application2.java]: provide in argument the first letter for filtering

Comment (acherkas) [inline Lecture2/task2/Application2.java]: there is no need to give name noFilterChildren, just set the name of list

File: Lecture2/task2/Application2.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                lastName = "Simpson";
+            }
+
+            Child child = new Child(k, "Tom" + k, lastName + k, age);
+            children.add(child);
+        }
+        return children;
+    }
+
+    public static Map<Integer, String> filteredById(Map<Integer, String> notFilteredMap) {
```
Comment (acherkas) [inline Lecture2/task2/Application2.java]: provide argument with key for filtering

File: Lecture2/task2/Application2.java (hunk 8 above)
Comment (acherkas) [inline Lecture2/task2/Application2.java]: extract method to Helper

File: Lecture2/task2/Application2.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        List<Child> filteredChildren = new ArrayList<>();
+        for (Child i : notFilteredChildren) {
+            if (i.getLastName().startsWith("S")) {
+                filteredChildren.add(i);
+            }
+        }
+        return filteredChildren;
+    }
+
+    public static void calculateAverageAgeOfChildren(List<Child> children) {
```
Comment (acherkas) [inline Lecture2/task2/Application2.java:31]: extract method to Helper

File: Lecture2/task2/Application2.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        Double average = children.stream().mapToInt(Child::getAge).average().getAsDouble();
+        System.out.println(average);
+    }
+
+    public static Map<Integer, String> convertListToMap(List<Child> listForConverting) {
```
Comment (acherkas) [inline Lecture2/task2/Application2.java:40]: extract method to Helper

File: Lecture2/task2/Application2.java (hunk 7 above)
Comment (acherkas) [inline Lecture2/task2/Application2.java]: extract method to Helper

File: Lecture2/task2/Application2.java (hunk 9 above)
Comment (acherkas) [inline Lecture2/task2/Application2.java]: extract method to Helper

File: Lecture2/task4/ComparingOfLists.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+public class ComparingOfLists {
```
Comment (acherkas) [inline Lecture2/task4/ComparingOfLists.java:7]: doesn't work properly

File: Lecture2/task1/Vegetable.java (hunk 1 above)
Comment (LeraShagoyko) [reply Lecture2/task1/Vegetable.java]: fixed

File: Lecture2/task1/Vegetable.java (hunk 2 above)
Comment (LeraShagoyko) [reply Lecture2/task1/Vegetable.java:4]: added

File: Lecture2/task1/Vegetable.java (hunk 3 above)
Comment (LeraShagoyko) [reply Lecture2/task1/Vegetable.java:13]: added

File: Lecture2/task1/Salad.java (hunk 4 above)
Comment (LeraShagoyko) [reply Lecture2/task1/Salad.java]: changed Map to ArrayList

File: Lecture2/task1/Vegetable.java (hunk 5 above)
Comment (LeraShagoyko) [reply Lecture2/task1/Vegetable.java]: changed to private

File: Lecture2/task2/Child.java (hunk 6 above)
Comment (LeraShagoyko) [reply Lecture2/task2/Child.java]: changed to private

File: Lecture2/task2/Application2.java (hunk 7 above)
Comment (LeraShagoyko) [reply Lecture2/task2/Application2.java]: done

File: Lecture2/task2/Application2.java (hunk 8 above)
Comment (LeraShagoyko) [reply Lecture2/task2/Application2.java]: done

Comment (LeraShagoyko) [reply Lecture2/task2/Application2.java]: why? I tried to use "speaking" name

File: Lecture2/task2/Application2.java (hunk 9 above)
Comment (LeraShagoyko) [reply Lecture2/task2/Application2.java]: done

File: Lecture2/task2/Application2.java (hunk 10 above)
Comment (LeraShagoyko) [reply Lecture2/task2/Application2.java:31]: I don't know yet what is Helper

**Pull request merged**